content/docs/rfcs/75/README.md

```diff
@@ -0,0 +1,173 @@
+
+### GetIntermediateNodeRequest
+
+This message is sent by the node that wants to sync with the peer node. It contains the node id/key and the level of the Prolly tree, uisng this information peer node searches and in response send back the subjected node. The peer node if using the Sync store protocol will respond with the GetIntermediateNodeResponse message. It sends the node of the tree to the requesting node.
+
+### GetIntermediateNodeResponse
+
+This message is sent by the peer node in response to the GetIntermediateNodeRequest message. It contains an intermediate node of the tree. The requesting node can then use this node to traverse the tree and find the missing nodes.
+
+### GetNonBoundaryNodesRequest
+
+This message is sent by the node that wants to sync with the peer node. It contains the start nodes from which the peer node will start searching for the non boundary nodes. The peer node if using the Sync store protocol will respond with the GetNonBoundaryNodesResponse message. It sends the non boundary nodes of the tree to the requesting node.
+
+### GetNonBoundaryNodesResponse
+
+This message is sent by the peer node in response to the GetNonBoundaryNodesRequest message. It contains the non boundary nodes of the tree. The requesting node can then use this nodes to traverse the tree and find the missing nodes.
+
+# Implementation in of Sync using Prolly tree (PoC version)
+
+This Section describes a proof-of-concept (PoC) implementation of the Sync protocol using Prolly tree.
+
+The PoC code is written in Python and can be found [here](https://github.com/ABresting/Prolly-Tree-Waku-Message). One can also run the code using the README.md guide.
+
+The Sync Process works in following steps:
+
+Step 1: A key-value store is created using the Store protocol where the key is the timestamp of a message, and the value is the `message_hash` attribute which uniquely identifies a message in Waku network.
+
+Step 2: A Prolly tree is populated using the key-value store. This construction is similar at both client and server side.
+
+Step 3: A client node sends a request to the server node to get the root node of the Prolly tree. The server node responds with the root node of the Prolly tree.
+
+Step 4: As Prolly tree is designed to compare diff based on equal hieght so if the height of the root of the Prolly tree is different then the client node sends a request to the server node to get the root node of the Prolly tree at the height of the root of the client's Prolly tree. The server node responds with the root node of the Prolly tree at the height of the root of the client's Prolly tree.
+
+Step 5: The client node then traverses the Prolly tree from the root node to the leaf nodes and finds the missing nodes at every level of the tree starting from the top. The missing intermediate Prolly tree Node is requested by client to the server. Server responds with the requested nodes.
```

Comment:
Is it possible to expand this step and explain the messages, field and logic involved:
e.g. if the client detects a difference in root, it requests the children of that node using message_x. For each child node that differs, it continues using message_x to retrieve child nodes until...